tests/array.rs

```diff
@@ -1970,6 +1970,77 @@ fn array_macros() {
     assert_eq!(empty2, array![[]]);
 }
 
+#[cfg(test)]
+mod as_standard_layout_tests {
```

Comment:
A lot of duplicate code can be removed from these tests with the following function:

```rust
fn test_as_standard_layout_for<S, D>(orig: ArrayBase<S, D>)
where
    S: Data,
    S::Elem: Clone + Debug + PartialEq,
    D: Dimension,
{
    let orig_is_standard = orig.is_standard_layout();
    let out = orig.as_standard_layout();
    assert!(out.is_standard_layout());
    assert_eq!(out, orig);
    assert_eq!(orig_is_standard, out.is_view());
}
```

Then, the body of each test is just calling `test_as_standard_layout_for` with an array.